packages/core/components/NestedDropdown/NestedDropdown.tsx

```diff
@@ -0,0 +1,251 @@
+
+                {tierTwo}
+              </li>
+            )
+          })}
+        </ul>
+      </li>
+    </>
+  )
+}
+
+interface NestedDropdownProps {
+  data: Record<string, string | number>[]
+  tiers: [string, string] // index 0  is the parent index 1 is the child
+  listLabel: string
+  handleSelectedItems: Function
+}
+
+const NestedDropdown: React.FC<NestedDropdownProps> = ({ data, tiers: [firstTierLabel, secondTierLabel], listLabel, handleSelectedItems }) => {
+  const optsMemo: Record<string, (string | number)[]> = {}
+
+  data.forEach(value => {
+    const tierOne = value[firstTierLabel]
+    const tierTwo = value[secondTierLabel]
+    if (optsMemo[tierOne]) {
+      optsMemo[tierOne].push(tierTwo)
+    } else {
+      optsMemo[tierOne] = [tierTwo]
+    }
+  })
+
+  const [userSelectedTierTwoLabel, setUserSelectedTierTwoLabel] = useState(null)
+  const [userSearchTerm, setUserSearchTerm] = useState('')
+  const [inputValue, setInputValue] = useState('')
+  const [inputHasFocus, setInputHasFocus] = useState(false)
+  const [isListOpened, setIsListOpened] = useState(false)
+
+  const searchInput = useRef(null)
+  const searchDropdown = useRef(null)
+
+  const chooseSelectedSecondTier = (tierOne: string, tierTwo: string) => {
+    searchInput.current.focus()
+    const selectedItemValue = tierTwo
+    setUserSelectedTierTwoLabel(tierOne + tierTwo)
+    setUserSearchTerm('')
+    setIsListOpened(false)
+    setInputValue(selectedItemValue)
+    handleSelectedItems(tierOne, tierTwo)
+  }
+
+  const handleKeyUp = e => {
+    const { nodeName, className, parentNode, nextSibling, lastChild, previousSibling } = e.target
+
+    switch (e.key) {
+      case 'ArrowDown': {
+        if (nodeName === 'INPUT') {
+          setIsListOpened(true)
+          // Move focus from Input to top of dropdown
+          searchDropdown.current.firstChild.focus()
+        } else if (className === 'selectableItem') {
+          // Move focus to next item on list: next Tier Two item or the next Tier One or SearchInput
+          const itemToFocusOnAfterKeyUp = nextSibling ?? parentNode.parentNode.nextSibling ?? searchInput.current
+          itemToFocusOnAfterKeyUp.focus()
+        } else if (className === 'nested-dropdown__group--hidden') {
+          // If Tier One is collapsed, move to next Tier One or move focus back to the top Input
+          const itemToFocusOnAfterKeyUp = nextSibling ?? searchInput.current
+          itemToFocusOnAfterKeyUp.focus()
+        } else {
+          // If Tier One is open, move focus to Tier Two
+          lastChild?.firstChild?.focus()
+        }
+        break
+      }
+
+      case 'ArrowUp': {
+        if (nodeName === 'INPUT') {
+          setIsListOpened(true)
+          // Move focus from Input to collapsed Tier Two or last item of Tier Two
+          if (searchDropdown.current.lastChild.className === 'nested-dropdown__group--hidden') {
+            searchDropdown.current.lastChild.focus()
```

Comment:
I'm not clear on what this is supposed to do...

maybe converting this thing to a proper <select> could mitigate some of these issues?